pkg/apis/events/v1alpha1/pubsub_source_types.go

```diff
@@ -51,14 +51,15 @@ var _ = duck.VerifyType(&PubSubSource{}, &duckv1alpha1.Conditions{})
 
 // PubSubSourceSpec defines the desired state of the PubSubSource.
 type PubSubSourceSpec struct {
-	// GcpCredsSecret is the credential to use to poll the GCP PubSubSource Subscription. It is not used
+	// Secret is the credential to use to poll the PubSubSource Subscription. It is not used
 	// to create or delete the Subscription, only to poll it. The value of the secret entry must be
 	// a service account key in the JSON format (see
 	// https://cloud.google.com/iam/docs/creating-managing-service-account-keys).
-	GcpCredsSecret corev1.SecretKeySelector `json:"gcpCredsSecret,omitempty"`
+	Secret *corev1.SecretKeySelector `json:"secret,omitempty"`
 
-	// GoogleCloudProject is the ID of the Google Cloud Project that the PubSubSource Topic exists in.
-	GoogleCloudProject string `json:"googleCloudProject,omitempty"`
+	// TODO: https://github.com/googleapis/google-cloud-go/blob/master/compute/metadata/metadata.go
```

Comment:
Not sure what the todo is here? Is it to support some kind of defaulting based off of that?